backend/src/test/java/gov/cdc/usds/simplereport/service/PatientBulkUploadServiceAsyncTest.java

```diff
@@ -0,0 +1,225 @@
+package gov.cdc.usds.simplereport.service;
+
+import static gov.cdc.usds.simplereport.api.Translators.parsePhoneType;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import gov.cdc.usds.simplereport.api.graphql.BaseGraphqlTest;
+import gov.cdc.usds.simplereport.db.model.Facility;
+import gov.cdc.usds.simplereport.db.model.Person;
+import gov.cdc.usds.simplereport.db.model.PhoneNumber;
+import gov.cdc.usds.simplereport.db.model.auxiliary.PersonRole;
+import gov.cdc.usds.simplereport.db.model.auxiliary.PhoneType;
+import gov.cdc.usds.simplereport.db.repository.PhoneNumberRepository;
+import gov.cdc.usds.simplereport.test_util.SliceTestConfiguration;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.test.context.TestPropertySource;
+
+/*
+ * We can't use the standard BaseServiceTest here because this service is async and requires a request context to operate.
+ * BaseFullStackTest doesn't have the authorization setup required for an authenticated test, but BaseGraphqlTest does.
```

Comment:
I wonder if we should consider renaming these base tests at some point given this async service test is not at all related to graphql.

Reply:
I think the correct solution here might be for me to redefine all the useful bits of `BaseGraphQlTest` into a different base test 😄 I'm mostly using it for the setup/teardown properties and authentication it offers, but it does have some graphql-specific methods that I'm just ignoring. Then we could have something like

```
BaseAuthenticatedFullStackTest extends BaseFullStackTest
BaseGraphQlTest extends BaseAuthenticatedFullStackTest
PatientBulkUploadServiceAsyncTest extends BaseAuthenticatedFullStackTest
```
I'm going to wait to do that until the test suite is passing consistently, though 😄 (We're almost there!! It now passes the full suite locally but not in CI)

Reply:
That sounds like a good separate ticket 😜

Reply:
Haha good call! Here it is: https://github.com/CDCgov/prime-simplereport/issues/4848